---
title: "Between-Sample (Beta) Diversity of Microbes in Snail Gut Microbiomes at Three Temperatures"
author: "Daphne Garcia"
date: "2025-04-16"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Set up the Environment 

```{r rmd-start}
# What time did we start running this script? 
start_time <- Sys.time()

set.seed(238428)

pacman::p_load(tidyverse, devtools, phyloseq, patchwork, vegan, 
               install = FALSE)

source("code/colors.R")

source("code/functions.R")
```

INTERPRETATION #1: What is your scientific question and hypotheses? Do they relate to abundance? Presence/absence? Phylogeny?

My question is whether there will be a difference in the composition of the Snail gut microbiomes between three different temperatures (control, high temp, low temp). I predict that there will be a more even abundance at control and lower temperatures, and that some species will rise to high dominance at high temperature. However, I predict that there will be more species present at control and lower temperatures than at high temperatures.
H0: There will be no change in microbial composition of snail gut microbiomes across three temperatures (high temperature, low temperature, control temperature)

HA: Microbial composition will differ in gut microbiomes at high temperature compared to low/control temperatures.

HA2: There will be significantly less species present in high temperature samples compared to low/control samples

# Load data
```{r load data}
# load phyloseq object
load("data/04_PhylogeneticTree/phytree_preprocessed_physeqs.RData")
midpoint_rooted_physeq

# Intuition check on seq depth
min(sample_sums(midpoint_rooted_physeq))


######## REMOVE TWO OUTLIERS: Unnecessary
# midpoint_rooted_physeq <- 
#   midpoint_rooted_physeq %>%
#   subset_samples(names != "SRR19509114") %>%
#   subset_samples(names != "SRR19509116") %>%
#   # And remove any ASVs unique to this sample, just in case.
#   prune_taxa(taxa_sums(.) > 0, .)
# 
# # Check that we now have 88 samples 
# midpoint_rooted_physeq
############################



# Create Metadata_df 
metadata_df <- 
  midpoint_rooted_physeq %>%
  sample_data() %>%
  data.frame()

head(metadata_df)
```

## Mutate metadata to have snail info
```{r mutate snail info}
## ADD A COLUMN FOR EXPEIMENTAL GROUP, SEX, AND TEMPERATURE

metadata_df <- metadata_df %>%
  mutate(snail_exp_group_name = substr(Sample.Name, 1, 2)) %>%
  mutate(temperature = substr(Sample.Name, 1, 1)) %>%
  mutate(sex = substr(Sample.Name, 2, 2))

```

# Normalizing the Read Depth

## 2. Explore the Raw Read Counts

```{r explore read counts}
# calculate read counts per sample 
raw_TotalSeqs_df <- 
  midpoint_rooted_physeq %>%
  # Calculate the total number of sequences/reads
  sample_sums() %>%
  data.frame()

# Take a look 
head(raw_TotalSeqs_df)
```

```{r explore read counts 2}
# Rename the column 
colnames(raw_TotalSeqs_df)[1] <- "TotalSeqs"

# add a new column of num_ASVs (RAW, non-noramlized # ASVs)
raw_TotalSeqsASVs_df <- 
  raw_TotalSeqs_df %>%
  mutate(num_ASVs = colSums(otu_table(midpoint_rooted_physeq) > 1))

#View(raw_TotalSeqsASVs_df)

# Plot histogram of seq depth 
rawSeq_histogram <- 
  raw_TotalSeqsASVs_df %>%
  ggplot(aes(x = TotalSeqs)) + 
  geom_histogram(bins = 50) + 
  scale_x_continuous(limits = c(30000, 80000)) + #changed limits
  labs(title = "Raw Seq Depth Histogram") + 
  theme_bw()

# Plot Seq Depth versus num_ASVs
rawSeq_vs_numASV_plot <- 
  raw_TotalSeqsASVs_df %>%
  ggplot(aes(x = num_ASVs, y = TotalSeqs)) + 
  geom_point() + 
  scale_y_continuous(limits = c(30000, 80000)) + ## changed limits
  geom_smooth(method = "lm") + 
  labs(title = "Seq Depth vs # ASVs") + 
  theme_bw()

# is it significant? 
summary(lm(TotalSeqs ~ num_ASVs, data = raw_TotalSeqsASVs_df))

# Put the two plots together.
rawSeq_histogram + rawSeq_vs_numASV_plot +
  plot_annotation(tag_levels = "A")
```

Summary of Seq depth vs num_ASVs
Multiple R-squared:  0.1386,	Adjusted R-squared:  0.1237 
F-statistic: 9.331 on 1 and 58 DF,  p-value: 0.003401


INTERPRETATION #2: Evaluating the above plots, what did you scale the read counts to in your dataset? Why did you choose that number? Do you think it is sufficient to perform a beta-diversity analysis and accurately compare your data to one another?

I am scaling to my minimum read count across samples, which is 39242. When statistically checking the relationship between the sequencing depth and number of ASV's, the R^2 is 0.139, which indicates that only 13.9% of the variation in ASV count is explained by sequencing depth. Since this is a weak correlation, it suggests that there is a weak correlation between sequencing depth and ASVs, which might mean that beta diversity metrics won't be overly skewed by sequencing depth.

The p value is 0.003, which indicates that the relationship is significant, so I didn't need to remove any outliers. Therefore, using the minimum read counts, 39k, was sufficient as a scale. Additionally, the raw sequencing depth histogram appears to have a normal distribution. Based on these points, I think that scaling to the minimum reads across samples, 39242, is  enough to perform beta diversity analysis.

# Scaling read counts
```{r scaling read counts}
min(sample_sums(midpoint_rooted_physeq))

# Scale the reads 
scaled_physeq <- 
  midpoint_rooted_physeq %>%
  scale_reads(round = "matround")

sample_data(scaled_physeq) <- metadata_df

# Look at it 
scaled_physeq
```

```{r look at scaling more}
# Look at it more!
#View(data.frame(otu_table(midpoint_rooted_physeq)))  # old, unscaled
#View(data.frame(otu_table(scaled_physeq)))           # new, scaled
  
# Confirm seq read depth of scaled_physeq 
scaled_TotalSeqs_df <- 
  scaled_physeq %>%
  sample_sums() %>%
  data.frame()

colnames(scaled_TotalSeqs_df)[1] <- "TotalSeqs"

# add a new column of num_ASVs (RAW, non-noramlized # ASVs)
scaled_TotalSeqsASVs_df <- 
  scaled_TotalSeqs_df %>%
  mutate(num_ASVs = colSums(otu_table(scaled_physeq) > 1))

# Scaling always gives us a range! 
# MInimum sequences
min(scaled_TotalSeqs_df$TotalSeqs) 
# Total Range 
range(scaled_TotalSeqs_df)

# How many is that range? 
range_seqs <- (max(scaled_TotalSeqs_df$TotalSeqs) - min(scaled_TotalSeqs_df$TotalSeqs))
range_seqs

# And in terms of percent? 
range_seqs/max(scaled_TotalSeqs_df$TotalSeqs)
```
the new minimum is 39159, new max is 39338. range of 179, which is 0.04% of the scaled total seqs

```{r plotting scaled seqs}

# plot the histogram of scaled sequence depth (moved from a r previous chunk)
# Plot it! 
scaledSeq_histogram <- 
  scaled_TotalSeqs_df %>%
  ggplot(aes(x = TotalSeqs)) + 
  geom_histogram(bins = 50) + 
  labs(title = "Scaled Seq Depth Histogram") + 
  scale_x_continuous(limits = c(35000, 40000)) + 
  theme_bw()

#View(scaled_TotalSeqs_df)

# Set the axis min and max for the next plot 
max_y <- max(scaled_TotalSeqs_df$TotalSeqs) + 10
min_y <- min(scaled_TotalSeqs_df$TotalSeqs) - 10

# Now, draw the plot 
scaledSeq_vs_numASV_plot <- 
  scaled_TotalSeqsASVs_df %>%
  ggplot(aes(x = num_ASVs, y = TotalSeqs)) + 
  geom_point() + 
  scale_y_continuous(limits = c(min_y, max_y)) +
  #geom_smooth(method = "lm") + 
  theme_bw() + 
  labs(title = "Scaled: Seq Depth vs # ASVs")

# Put the two plots together.
scaledSeq_histogram + scaledSeq_vs_numASV_plot +
  plot_annotation(tag_levels = "A")

```

INTERPRETATION #3: Evaluating the scaled read count plots above, what did you scale the read counts to in your dataset? Why did you choose that number? Do you think it is sufficient to perform a beta-diversity analysis and accurately compare your data to one another?

I scaled using the raw mininum reads per sequence, which was 39,242. After scaling, the new minimum is 39159, and new max is 39338. There is a range of 179, which is 0.04% of the scaled total seqs. Looking at the scaled plots, I feel confident that I scaled sufficiently, because there does not seem to be a correlation between sequencing depth and number of ASVs anymore, and the scaled sequence depth histogram is a very narrow range around 39000 as expected.

# Calculating dissimilarity
```{r calc dissimilarity}

# Sorensen Dissimiliarty
scaled_sorensen_dist <- phyloseq::distance(scaled_physeq, method = "bray", binary = TRUE)

# What does it look like? 
class(scaled_sorensen_dist)

str(scaled_sorensen_dist)

```
INTERPRETATION #4: Which dissimilarity metric is most related to your scientific question? Why or why not?

I think weighted unifrac is best suited to both my alternate hypotheses from the first interpretation. I have two main questions: Whether the abundance will differ between the hot-temperature gut communities and the other communitites (lower-temperature and control) and additionally whether there are unique dominant taxa in the high-temperature gut communitites compared to the other groups. These two questions aim to validate the original paper's results, which found that higher temperature samples had lower abundances and high dominance of opportunistic pathogens. For the first question of abundances, a Bray Curtis test fits because it weighs the abundances of the different ASVs in the experimental groups. However, Weighted UniFrac will best answer both the first question, which only focuses on abundances, and the second question, which incorporates abundance and also the uniqueness of newly dominant taxons, which can rely on phylogeny to distinguish unique taxons.


```{r more calc dissimilarity}
#head(as.matrix(scaled_sorensen_dist))

# Bray-Curtis Dissimiliarty
scaled_bray_dist <- phyloseq::distance(scaled_physeq, method = "bray", binary = FALSE)

# Abundance-Unweighted UniFrac
scaled_uUnifrac_dist <- phyloseq::distance(scaled_physeq, method = "unifrac")

# Abundance-Weighted UniFrac
scaled_wUnifrac_dist <- phyloseq::distance(scaled_physeq, method = "wunifrac")
```


How interpret sorensen?
- in class ex: test station, date, both. by = terms, will evaluate station first, and then look at date.
first look at r^2, f, and p value. we see r^2 is highest for station, then interaction station-date, then just date. by those 3 terms alone, can explain 60% (all of this is told by blurb). date is important but it explains less of variation.
how to interpret a flipped model? it shows difference in num of variation explained

# PERMANOVA

## Sorensen 
```{r sorensen}
# Sorensen
## 1. Run with by = terms for R² values, sensitive to order of variables! 
## ALWAYS check and confirm the order of your terms and how they interact with each other.  
sorensen_station_adonis_terms1 <- adonis2(scaled_sorensen_dist ~ temperature * sex, data = metadata_df, by = "terms")
sorensen_station_adonis_terms1
```

                Df SumOfSqs      R2      F Pr(>F)    
temperature      2   1.2968 0.07356 2.3482  0.001 ***
sex              1   0.6358 0.03606 2.3026  0.001 ***
temperature:sex  2   0.7865 0.04461 1.4242  0.016 *  
Residual        54  14.9111 0.84577                  
Total           59  17.6303 1.00000                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

R^2 are low, 0.07, 0.03, and 0.04. This means almost none (only 15%) of the variance in the data are explained by temperature or sex. For temp and sex separately, P value is under 0.001, which means the groups within sex and temperature are significantly different from one another. But a low F value (2.3 for temp and sex) means that though the groups within temp and sex are significantly different, they do not differ by much.

```{r sorensen change order}

# Check the order of the terms
sorensen_station_adonis_terms2 <- adonis2(scaled_sorensen_dist ~ sex * temperature, data = metadata_df, by = "terms")
sorensen_station_adonis_terms2

# Check only one term, that encapsulates both temp and sex
sorensen_station_adonis_terms3 <- adonis2(scaled_sorensen_dist ~ snail_exp_group_name, data = metadata_df, by = "terms")
sorensen_station_adonis_terms3

```

Changing the order, and lumping them into one variable changes nothing for R^2, but the P values are higher, which means the groups are less significantly different now. But F value stays the same (2.3)

```{r soresen permanova}
## 2. Run with by = "margin" for marginal p-values, which we can compare to the residuals from the first one. 
sorensen_station_adonis_margin <- adonis2(scaled_sorensen_dist ~ temperature * sex, data = metadata_df, by = "margin")
sorensen_station_adonis_margin
```

now it only shows the temperature:sex, which has same R^2 the same as when using "by: term", but F value is much lower, and p value much higher, which means that separating by term instead of margin is best.

## Bray Curtis


Bray curtis: the only thing that changes from sorensen, is abundance value

Since the bray curtis and sorensen were very similar, this means abundance doesn't tell us about the changes over time for the variables that have same bray curtis and sorensen. So, the other variable that WAS affected differently by sorensen and bray, probably has a higher effect on changing community composition, and also exclusively affects community composition
```{r}
# Bray-Curtis
## ALWAYS check and confirm the order of your terms and how they interact with each other.  
bray_station_adonis_terms1 <- adonis2(scaled_bray_dist ~ temperature * sex, data = metadata_df, by = "terms")
bray_station_adonis_terms1
```
                 Df SumOfSqs      R2       F Pr(>F)    
temperature      2   3.3506 0.25234 10.3073  0.001 ***
sex              1   0.4923 0.03708  3.0289  0.005 ** 
temperature:sex  2   0.6582 0.04957  2.0249  0.013 *  
Residual        54   8.7770 0.66101                   
Total           59  13.2782 1.00000                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
 
MUCH higher than sorensen. R^2 for temp (when accounting for abundance va bray curtis) explains 25% of temperature, and now the F statistic is higher. The sex R^2 is higher, but p value and f statistic are worse.

```{r more bray}

bray_station_adonis_terms2 <- adonis2(scaled_bray_dist ~ sex * temperature, data = metadata_df, by = "terms")
bray_station_adonis_terms2

```

Flipping temp and sex yeilds the exact same R^2 and F, but slightly better p values

                Df SumOfSqs      R2       F Pr(>F)    
sex              1   0.4923 0.03708  3.0289  0.003 ** 
temperature      2   3.3506 0.25234 10.3073  0.001 ***
sex:temperature  2   0.6582 0.04957  2.0249  0.010 ** 
Residual        54   8.7770 0.66101                   
Total           59  13.2782 1.00000   

```{r bray flip margin}

## 2. Run with by = "margin" for marginal p-values
bray_station_adonis_margin <- adonis2(scaled_bray_dist ~ sex * temperature, data = metadata_df, by = "margin")
bray_station_adonis_margin

```
margin gives same R^2 and bad f and p values.


Interpreting PERMANOVA:
By looking at both soresen and bray curtis, temperature is the strongest factor, explaining 7% of the variation in microbial community composition in soresen and a much higher 25% of variation in bray curtis, which is statistically significant based on the p-value of <0.001. This suggests that the temperatures at which snails are in affects the community structure of their gut microbiomes.
Sex contributes much less, yet still significantly (R^2 = 3%, p = 0.003). This suggests that the sex of the snail host lightly impacts the community composition

The interaction between sex and temperature explains an additional 4% of the variation, which is around the same as much as sex explains alone. with a significant pvalue of <0.01, which is a much higher p value than temperature and sex alone. Therefore, temp and sex do not generally act together to change the composition of snail gut microbiomes.
The total residiuals of the model mean that ~66% of the variation remains unexplained by these variables, which is fairly reasonable. 

The Bray-Curtis model explains ~20% more total variation (from 84% residual to 66% residual), suggesting that abundance plays a large role in affecting the community composition of gut microbiomes in this study.
The higher R^2 for temperature specifically in Bray-Curtis (~25% vs. ~7%) indicates that abundance shifts across different temperatures are even more pronounced than presence/absence alone would suggest. Additionally, the p and f values improve in the bray-curtis model, further supporting that abundance, not just species turnover, is important in structuring community structure of these microbiomes across different temperatures and sex of the hosts.


# Beta Disper

betadisper: we were assuming equal distances between variances.. so now we need to test to make sure this is true
- run betadisper
- we may have a biological reason related to variances
run sorensen betadisper: see if you fit homogeneity assumption for "date/station" or other variables.

## soresen betadisper
```{r beta disper soresen}

# Homogeneity of Disperson test with beta dispr
# Sorensen Betadisper - Temperature
dispr_sorensen_temperature <- betadisper(scaled_sorensen_dist, metadata_df$temperature)
# permutest() performs a non-parametric permutation test, which is robust and valid for the kind of data used in beta diversity analysis (e.g., dissimilarity matrices).
permutest(dispr_sorensen_temperature)

# Sorensen Betadisper - sex  
dispr_sorensen_sex <- betadisper(scaled_sorensen_dist, metadata_df$sex)
permutest(dispr_sorensen_sex)
```
TEMPERATURE

          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)  
Groups     2 0.023432 0.0117161 4.1296    999  0.023 *
Residuals 57 0.161714 0.0028371                       
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

SEX
          Df   Sum Sq   Mean Sq     F N.Perm Pr(>F)
Groups     1 0.001386 0.0013857 0.486    999  0.488
Residuals 58 0.165365 0.0028511 

Interpretint Betadisper sorensen:

## Bray Curtis betadisper
```{r beta disper bray}

# Bray-Curtis Betadisper - Station 
dispr_bray_temperature <- betadisper(scaled_bray_dist, metadata_df$temperature)
permutest(dispr_bray_temperature)

# Bray-Curtis Betadisper - sex 
dispr_bray_sex <- betadisper(scaled_bray_dist, metadata_df$sex)
permutest(dispr_bray_sex)

```

TEMPERATURE
          Df  Sum Sq  Mean Sq      F N.Perm Pr(>F)   
Groups     2 0.19851 0.099256 5.0859    999  0.007 **
Residuals 57 1.11240 0.019516                        
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

SEX
          Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)
Groups     1 0.00923 0.0092298 0.4903    999  0.529
Residuals 58 1.09192 0.0188262  


INTERPRETATION #6
Interpreting beta disper of Bray Curtis:

Temperature has a significant betadisper()/permutest() result (p = 0.023). Therefore, we cannot conclude for certain that our PERMANOVA results are truly because of temperature alone. I did not expect there to be dispersion within a temperature group, as I thought the same species would be present in all high-temperature guts. However, maybe in different hosts, high temperature leads to the rise of new species independent of others under the same high temperature. Sex does not have a significant betadisper()/permutest() result (p = 0.488). Therefore, the PERMANOVA results likely represent a true difference by sex.

The Bray-Curtis results are in agreement with the Sorensen results for Temperature (p = 0.023 vs p = 0.007) and sex ( p = 0.488 vs p = 0.529) So, we can conclude that temperature likely has some biological difference in the variance, which means that the PERMANOVA interpretations aren't as straightforward. Sex, however, does not have heterogeneity of variances, meaning that the PERMANOVA results are likely accurate.

INTERPRETATION #7

Since the sex permanova was significant, yet betadisper wasn't, we can assume a valid permanova result. Therefore, there are likely true compositional differences between the gut microbiomes of male and female snails.
However, since the temperature permanova was significant, and ALSO betadisper was, then the differences between categorical termperature groups may be due to dispersion, not centroids. We cannot assume, based on permanova alone, that temperature is cause for compositional differences of snail gut microbiome communities between experimental groups.

# PCoA and NMDS plots

## PCoA with Soresen
```{r pcoa soresen}

### SORENSEN 
# First, calculate PCoA with Soresen
scaled_soren_pcoa <- 
  ordinate(physeq = scaled_physeq,
         method = "PCoA",
         distance = "bray", binary = TRUE)

# Take a quick look
str(scaled_soren_pcoa)

# Plot it: Sorensen PCoA  
sorensen_pcoa_plot <- 
  plot_ordination(physeq = scaled_physeq,
                ordination = scaled_soren_pcoa,
                color = "temperature",
                shape = "sex",
                title = "Sorensen PCoA") + 
  scale_color_manual(values = temperature_colors) + 
  scale_shape_manual(values = c(15, 16, 17)) + 
  geom_point(size = 5, alpha = 0.5, aes(color = temperature)) + 
  theme_bw() + 
  theme(legend.position = "right")

```

### PCoA with bray curtis
```{r bray curtis pcoa}
### Bray-Curtis 
# Second, calculate PCoA with Bray-Curtis
scaled_bray_pcoa <- 
  ordinate(physeq = scaled_physeq,
         method = "PCoA",
         distance = "bray", binary = FALSE)

# Plot it: Bray-Curtis PCoA 
bray_pcoa_plot <- 
  plot_ordination(physeq = scaled_physeq,
                ordination = scaled_bray_pcoa,
                color = "temperature",
                shape = "sex",
                title = "Bray-Curtis PCoA") + 
  scale_color_manual(values = temperature_colors) + 
  scale_shape_manual(values = c(15, 16, 17)) + 
  geom_point(size = 5, alpha = 0.5, aes(color = temperature)) + 
  theme_bw() + 
  theme(legend.position = "right")

# Show the plots 
sorensen_pcoa_plot + bray_pcoa_plot + 
  plot_layout(guides = "collect") +
  plot_annotation(tag_levels = "A")
```

 INTERPRETATION #8
From the plots of the Sorensen PCoA and Bray Curtis PCoA plots, we can conclude that temperature is the dominant factor structuring microbial communities, as all of the different colors (temperatures) form individual groups in both the sorensen and bray curtis plots. Sex of the host snail, reflected in the shapes, appears to factor into axis 1 of the sorensen plot, yet may not explain as much of the variation as temperature

Bray-Curtis highlights both temperature and sex patterns more strongly because between the two axes, more variation is explained (~41%) than Sorensen (~20%). The 21% gap between the PCoA plots supports previous PERMANOVA and Bray dissimilarity method results that show the abundance differences among experimental groups matter in snail gut microbiomes. 

# NMDS plots

 NMDS works to best match the original distance rankings from a dissimilarity matrix, such as Bray-Curtis. The quality of the fit is measured by strlower stress. Lower stress indicates a better match.
Stress < 0.1 is strong
Stress < 0.15 is accepted but 
Sress < 0.2 is the limit of accepted.
 
 NMDS does not assume linear relationships and is less prone to artifacts, so it is better for exploring patterns in complex microbial datasets.

## sornesen
```{r nmds sorensen}
## SORENSEN 
scaled_soren_nmds <- 
  ordinate(physeq = scaled_physeq,
         method = "NMDS",
         distance = "bray", binary = TRUE)

sorensen_nmds_plot <- 
  plot_ordination(physeq = scaled_physeq,
                ordination = scaled_soren_nmds,
                color = "temperature",
                shape = "sex",
                title = "Sorensen NMDS") + 
  scale_color_manual(values = temperature_colors) + 
  scale_shape_manual(values = c(15, 16, 17)) + 
  geom_point(size = 5, alpha = 0.5, aes(color = temperature)) + 
  theme_bw() + 
  theme(legend.position = "right")

```

## bray curtis nmds
```{r nmds bray}

### Bray-Curtis 
# Second, calculate NMDS with Bray-Curtis
scaled_bray_nmds <- 
  ordinate(physeq = scaled_physeq,
         method = "NMDS",
         distance = "bray", binary = FALSE)

# Plot it: Bray-Curtis NMDS 
bray_nmds_plot <- 
  plot_ordination(physeq = scaled_physeq,
                ordination = scaled_bray_nmds,
                color = "temperature",
                shape = "sex",
                title = "Bray-Curtis NMDS") + 
  scale_color_manual(values = temperature_colors) + 
  scale_shape_manual(values = c(15, 16, 17)) + 
  geom_point(size = 5, alpha = 0.5, aes(color = temperature)) + 
  theme_bw() + 
  theme(legend.position = "right")

# Show the plots 
sorensen_nmds_plot + bray_nmds_plot + 
  plot_layout(guides = "collect") +
  plot_annotation(tag_levels = "A")

```
INTERPRETATION #9: Plot your data using NMDS. Interpret what you can conclude from the NMDS.

Looking at the combined plot of Sorensen and Bray-Curtis NMDS plots, they both look extremely similar, just flipped horizontally and rotated 90%. I find it interesting that for NMDS, the abundance-independent and abundance-weighted methods appear to place data points in same places respective to one another, considering that for all other graphs there was a clear distinction between sorensen and bray curtis. Therefore, biologically this could mean that in the scheme of many variables, abundance does not account for much difference between experimental groups. Additionally, the colors (different temperature groups) and shapes (sex of host: M or F), do not form clear, separate clusters, which indicates that temperature and sex do not have a great effect on the order of differences between community structures of snail gut microbiomes.Looking at the stress of NMDS, for sorensen: stress for all 20 runs was between 0.14 and 0.21 for bray curtis: stress for all 20 runs was between 0.16 and 0.24. All stresses are either on the cusp of being acceptable or barely acceptable. 


## All ordinations together:
```{r pcoa and nmds}
sorensen_pcoa_plot + bray_pcoa_plot + 
sorensen_nmds_plot + bray_nmds_plot + 
  plot_layout(guides = "collect") +
  plot_annotation(tag_levels = "A")
```

INTERPRETATION #10:
The Bray Curtis PCoA plots best represents the data from this study: When comparing PCoA plots to NMDS plots, PCoA shows clear separation between temperature and host-sex experimental groups, whereas NMDS overlaps these groups with no distinct clustering. In the PCoA plot, the separation of experimental groups into clusters in the Bray-Curtis plot reveal stronger separation of variables, suggesting that abundance differences have a significant role in the snail gut microbiome. This aligns with the PERMANOVA results and the initial Bray-Curtis statistics from earlier in the homework. Additionally, this plot has the highest percents of influence (41% between both axes) In contrast, NMDS shows minimal distinction between Sorensen and Bray-Curtis plots, which doesn't support the overall findings of abundance making a difference. Additionally, both NMDS plots had very high stress, indicating that the data did not fit well into 2 variables by order. While NMDS may be superior to PCoA plots for datasets with many outliers or complex data, this dataset does not fit that profile. Therefore, the Bray-Curtis PCoA plot provides the best representation of the data.
