---
title: "March1_briefwork_trimming_snail_data"
author: "Daphne Garcia"
date: "2025-03-02"
output: html_document
editor_options: 
  chunk_output_type: console
---

#Setting the seed
```{r set-seed}
# Any number can be chosen 
set.seed(238428)
```

#Setting start time
```{r set start time}
# What time did we start running this script? 
start_time <- Sys.time()
start_time
```

#Load Libraries
```{r load libraries}
# Efficient package loading with pacman 
# Don't forget to install pacman and DT if you don't have it yet. :) 
pacman::p_load(tidyverse, dada2, phyloseq, patchwork, DT, devtools, install = FALSE)
library("patchwork")
```


#Read RAW sequencing files + intuition check
```{r read raw sequencing files, intuition check}
# Set the raw fastq path to the raw sequencing files 
# Path to the fastq files 
raw_fastqs_path <- "data/data_raw"        ############################CHANGE

raw_fastqs_path

#intuition checks

head(list.files(raw_fastqs_path))
length(list.files(raw_fastqs_path))
```
_2.fastq.gz

#create forward/reverse reads + intuition check
```{r create forward/reverse reads}
# Create vector of forward reads
forward_reads <- list.files(raw_fastqs_path, pattern = "_1.fastq.gz", full.names = TRUE)  


# Create vector of reverse reads
reverse_reads <- list.files(raw_fastqs_path, pattern = "_2.fastq.gz", full.names = TRUE) 


# Intuition Checks 
head(forward_reads)
stopifnot(length(forward_reads) < length(list.files(raw_fastqs_path)))
head(reverse_reads)

stopifnot(length(reverse_reads) == length(forward_reads))
length(forward_reads) + length(reverse_reads)
```

## IGNORE: this was test to see if ncbi-downloaded files had quality score
```{r TESTING ncbi-downloaded snail data with F phred score}

# Create vector of forward reads
#forward_read <- list.files("../March3/sra", pattern = "_1.fastq.gz", full.names = TRUE)
#forward_filteredQual_plot_12 <- plotQualityProfile(forward_read)

# Create vector of reverse reads
#reverse_read <- list.files("../March3/sra", pattern = "_2.fastq.gz", full.names = TRUE) 
#reverse_filteredQual_plot_12 <- plotQualityProfile(reverse_read)

#forward_filteredQual_plot_12 + reverse_filteredQual_plot_12

```




#Assess Raw Read Quality
```{r Plotting PRE-QC plots}
# Randomly select 12 samples from dataset to evaluate 
# Selecting 12 is typically better than 2 (like we did in class for efficiency)
random_samples <- sample(1:length(reverse_reads), size = 12)
random_samples



# Calculate and plot quality of these two samples
forward_filteredQual_plot_12 <- plotQualityProfile(forward_reads[random_samples]) + 
  labs(title = "Forward Read: Raw Quality")

reverse_filteredQual_plot_12 <- plotQualityProfile(reverse_reads[random_samples]) + 
  labs(title = "Reverse Read: Raw Quality")

# Plot them together with patchwork
forward_filteredQual_plot_12 + reverse_filteredQual_plot_12
```



#Aggregated Raw Quality Plots
```{r Aggregated Raw Quality Plots}
# Aggregate all QC plots 
# Forward reads
forward_preQC_plot <- 
  plotQualityProfile(forward_reads, aggregate = TRUE) + 
  labs(title = "Forward Pre-QC")

# reverse reads
reverse_preQC_plot <- 
  plotQualityProfile(reverse_reads, aggregate = TRUE) + 
  labs(title = "Reverse Pre-QC")

# Now, let's put the two plots together
preQC_aggregate_plot <- 
  # Plot the forward and reverse together 
  forward_preQC_plot + reverse_preQC_plot
# Show the plot
preQC_aggregate_plot
```

INTERPRETATION #1 of Pre-QC:The majority of the reads have a score of over 30. The green is the mean, and orange lines are the median and quantiles. The orange lines are for the most part a solid 36, which means only rarest data (under 25th quantile) has lowering quality scores towards the ends of the reads.

High quality in early and mid-read: Both in the forward and reverse plots, the beginning and middle of the plots is solidly above 33. The reverse plot declines in quality at mid-sequence faster than the forward plot, which is expected, as the sequencing of the reverse reads happens second, allowing more degradation of the chemistry and there are more accumulated errors.



#Preparing placeholder for filtered reads
```{r}
# Create vector of sample names from the filenames 
sample_names <- sapply(strsplit(basename(forward_reads), "_"), `[`,1) 

# Intuition Check 
head(sample_names)

# Place filtered reads into filtered_fastqs_path
filtered_fastqs_path <- "data/data_filtered"

# Intuition Check 
filtered_fastqs_path

# create 2 vectors: filtered_forward_reads & filtered_reverse_reads
filtered_forward_reads <- 
  file.path(filtered_fastqs_path, paste0(sample_names, "_R1_filtered.fastq.gz"))

# Intuition Check 
length(filtered_forward_reads)

# reverse reads
filtered_reverse_reads <- 
  file.path(filtered_fastqs_path, paste0(sample_names, "_R2_filtered.fastq.gz"))

# Intuition Check 
head(filtered_reverse_reads)

length(filtered_reverse_reads)
```



Notes about trimming:
maxEE = "maximum error", online says "which is a better filter than simply averaging quality scores."
Since mine are mostly above 30, use maxEE (2,2)

Do the reads dip suddenly in their quality? If so, explore trimLeft and truncLen


#TRIMMING
```{r trimming step}

# Assign and write out filtered fastq files 
# Here, in this class dataset, the Kozich et al.(2013) AEM
      # Link to paper: https://doi.org/10.1128/AEM.01043-13
# Therefore, we do not need to trim the primers, because they were not sequenced
filtered_reads <- 
  filterAndTrim(fwd = forward_reads, filt = filtered_forward_reads, rev = reverse_reads, filt.rev = filtered_reverse_reads,
              maxN = 0, maxEE = c(2,2), 
              # Remove ZERO first bases at the beginning of the forward read, because all scores above 30
              trimLeft = 0,
              # Keep the full forward & trim final 3 bases in the reverse with truncLen, because there's small dip
              truncLen = c(250, 247), 
              truncQ = 2, rm.phix = TRUE, compress = TRUE, multithread = TRUE)


```


#Assess Trimmed Read quality

```{r}
# Plot the 12 random samples after QC
forward_filteredQual_plot_12 <- 
  plotQualityProfile(filtered_forward_reads[random_samples]) + 
  labs(title = "Trimmed Forward Read Quality")

reverse_filteredQual_plot_12 <- 
  plotQualityProfile(filtered_reverse_reads[random_samples]) + 
  labs(title = "Trimmed Reverse Read Quality")

# Put the two plots together 
forward_filteredQual_plot_12 + reverse_filteredQual_plot_12
```


#Aggregated Trimmed plots
```{r aggregated trimmed plots}
# Aggregate all QC plots 
# Forward reads
forward_postQC_plot <- 
  plotQualityProfile(filtered_forward_reads, aggregate = TRUE) + 
  labs(title = "Forward Post-QC")

# reverse reads
reverse_postQC_plot <- 
  plotQualityProfile(filtered_reverse_reads, aggregate = TRUE) + 
  labs(title = "Reverse Post-QC")

# Now, let's put the two plots together
postQC_aggregate_plot <- 
  # Plot the forward and reverse together 
  forward_postQC_plot + reverse_postQC_plot
# Show the plot
postQC_aggregate_plot
```

Interpretation of aggregate trimmed plots:
Forward Reads Post-QC: Mean sequence has high-quality bases above 30 throughout its entirety.There starts to be a decline around base 200, but still largely above 30. At around 225, there's one base that has a 25th percentile (orange dashed line) of around 25, but this is better compared to the raw read, where there were two dips. 
A slight decline after ~220-230 cycles but still largely above a Phred of 30.

Reverse Reads Post-QC: The reverse sequence also has a score of above 30 throughout the entire length. The initial bases are almost perfectly scored at 37, and there is a lot of stability mid-length until around 200. It is more stable than the raw reverse QC, which suggests successful trimming. Additionally, the end of the length stays at around 32, and no longer dips down like it did in the raw reads. 

Takeaway: Quality filtering successfully retained high-quality bases while removing low-quality reverse end tail.


# Read retention post QC
```{r}

# Make output into dataframe 
filtered_df <- as.data.frame(filtered_reads) %>%
  mutate(percent.retained = reads.out/reads.in)
# Intuition check
# Visualize it in table format 
DT::datatable(filtered_df)


# Let's calculate some statistics
read_stats_df <- 
  filtered_df %>%
  reframe(median_reads_in = median(reads.in),
          median_reads_out = median(reads.out),
          median_percent_retained = (median(reads.out)/median(reads.in)),
          max_percent_retained = max(reads.out/reads.in),
          min_percent_retained = min(reads.out/reads.in))

# Take a look at it!
read_stats_df

```


##Plotting read retention plots
```{r plotting read retention}

# Plot it 
numSeqs_QC_dotplot <-
  filtered_df %>%
  ggplot(aes(x = reads.in, y = reads.out)) + 
  geom_point(alpha = 0.5, size = 2) + 
  labs(x = "# of Raw Seqs", 
       y = "# of Seqs Retained") + 
  # Now let's add a 1:1 line for reference of keeping 100% of the reads
  geom_abline(slope=1, intercept = 0, color = "deeppink")

# Now, let's look at the number of reads retained in a histogram
numRetained_QC_histplot <- 
  filtered_df %>%
  ggplot(aes(x = reads.out)) + 
  geom_histogram() + 
  labs(x = "# of Seqs Retained", 
       y = "# of Samples") 

# Create a histogram of percent reads retained in a histogram
percSeqs_QC_histplot <- 
  filtered_df %>%
  ggplot(aes(x = percent.retained)) + 
  geom_histogram() + 
  labs(x = "% of Seqs Retained", 
       y = "# of Samples") + 
  # Set the scale to be between 0-1 (0-100%)
  scale_x_continuous(limits = c(0, 1))

# Now, let's put the plots together
numSeqs_QC_dotplot + numRetained_QC_histplot + percSeqs_QC_histplot + 
  plot_annotation(tag_levels = 'A')

```


Interpretation of sequence retention plots:

Panel A:
Points are not close to the diagonal, yet they match the slope of the pink line ("perfect retention line"). This means that Samples generally lost a proportional amount of sequences during the filtering. There are two outliners, which indicate that two samples did lose significantly more reads than the others. Overall, this plot indicates that a lot of sequences were lost, and the trimming was impactful, but an equal proportion was lost from samples which hopefully means that different lapha diversity estimates won't be too impacted.

Panel B: 
The majority of samples have between ~90,000 and 110,000 retained sequences, which suggests good filtering efficiency.
A few samples have lower retention, around 70,000 retained samples but the distribution is generally a bell curve, indicating a consistent range in read quality.

Panel C: Histogram of Percent of Sequences Retained

The vast majority of samples had over 80% of samples retained, with one having around 75% retained, which we can keep our eye on. We had a median 89.8% of samples retained, which is great. 

Overall, I think this run of filterAndTrim() is sufficient



## Visualize QC differences in plot
```{r}
preQC_aggregate_plot / postQC_aggregate_plot
```

Interpretation of aggregate pre vs post QC plots

Beginning of read: No bases at the beginning were trimmed, because in the raw QC plots,  phred scores stayed consistently around 37 until later mid-read in both forward and reverse plots. This stays true in the trimmed QC plots, which are still ~37.

Middle of read: The quality remains consistently around 37 most cycles/bases.

End of read: While a vertical dotted line still remains in the trimmed foward reads between mid- end of the read, the trimmed plot only has one, while the raw plot had two. The raw reverse read had a sharp drop in quality in the last three bases, with the 25th quantile dropping to ~25, and this is completely removed in the trimmed reverse read. Now, forward and reverse reads stay above ~32 throughout the whole read.


# End time and session info
```{r}
# Take the time now that we are at the end of the script
end_time <- Sys.time()
end_time 

devtools::session_info()
```



I trust that I have sufficiently QC'd my data. First, all the raw quality scores were above 30 in the first place, so there was little quality control to be done. I trimmed the end 3 bases of the reverse read, which took care of a sharp drop in quality score at the end of the reverse read, and coincidentally removed one of the two sharp drops of the 25th quantile from the forward QC plot that was in the middle of the plot. There wasn't perfect retention of the reads, as shown by the scatter plot of raw vs retained sequences, where the points were all under the pink "perfect retention line". However, the percent of sequences retained shows the vast majority of sequences retained over 80% of reads, which is a good sign that the reads were retained. Overall, I believe this is sufficient QC.
