---
title: "March1_briefwork_trimming_snail_data"
author: "Daphne Garcia"
date: "2025-03-02"
output: pdf_document
editor_options: 
  chunk_output_type: console
---




#Setting the seed
```{r set-seed}
# Any number can be chosen 
set.seed(238428)
```

#Setting start time
```{r set start time}
# What time did we start running this script? 
start_time <- Sys.time()
start_time
```

#Load Libraries
```{r load libraries}
# Efficient package loading with pacman 
# Don't forget to install pacman and DT if you don't have it yet. :) 
pacman::p_load(tidyverse, dada2, phyloseq, patchwork, DT, devtools, install = FALSE)
library("patchwork")
```


#Read RAW sequencing files + intuition check
```{r read raw sequencing files, intuition check}
# Set the raw fastq path to the raw sequencing files 
# Path to the fastq files 
raw_fastqs_path <- "../March3/sra"        ############################CHANGE

raw_fastqs_path

#intuition checks

head(list.files(raw_fastqs_path))
length(list.files(raw_fastqs_path))
```
_2.fastq.gz

#create forward/reverse reads + intuition check
```{r create forward/reverse reads}
# Create vector of forward reads
forward_reads <- list.files(raw_fastqs_path, pattern = "_1.fastq.gz", full.names = TRUE)  


# Create vector of reverse reads
reverse_reads <- list.files(raw_fastqs_path, pattern = "_2.fastq.gz", full.names = TRUE) 


# Intuition Checks 
head(forward_reads)
stopifnot(length(forward_reads) < length(list.files(raw_fastqs_path)))
head(reverse_reads)

stopifnot(length(reverse_reads) == length(forward_reads))
length(forward_reads) + length(reverse_reads)
```

## IGNORE: this was test to see if ncbi-downloaded files had quality score
```{r TESTING ncbi-downloaded snail data with F phred score}

# Create vector of forward reads
forward_read <- list.files("../March3/sra", pattern = "_1.fastq.gz", full.names = TRUE)
forward_filteredQual_plot_12 <- plotQualityProfile(forward_read)

# Create vector of reverse reads
reverse_read <- list.files("../March3/sra", pattern = "_2.fastq.gz", full.names = TRUE) 
reverse_filteredQual_plot_12 <- plotQualityProfile(reverse_read)

forward_filteredQual_plot_12 + reverse_filteredQual_plot_12

```




#Assess Raw Read Quality
```{r Plotting PRE-QC plots}
# Randomly select 12 samples from dataset to evaluate 
# Selecting 12 is typically better than 2 (like we did in class for efficiency)
random_samples <- sample(1:length(reverse_reads), size = 12)
random_samples



# Calculate and plot quality of these two samples
forward_filteredQual_plot_12 <- plotQualityProfile(forward_reads[random_samples]) + 
  labs(title = "Forward Read: Raw Quality")

reverse_filteredQual_plot_12 <- plotQualityProfile(reverse_reads[random_samples]) + 
  labs(title = "Reverse Read: Raw Quality")

# Plot them together with patchwork
forward_filteredQual_plot_12 + reverse_filteredQual_plot_12
```



#Aggregated Raw Quality Plots
```{r Aggregated Raw Quality Plots}
# Aggregate all QC plots 
# Forward reads
forward_preQC_plot <- 
  plotQualityProfile(forward_reads, aggregate = TRUE) + 
  labs(title = "Forward Pre-QC")

# reverse reads
reverse_preQC_plot <- 
  plotQualityProfile(reverse_reads, aggregate = TRUE) + 
  labs(title = "Reverse Pre-QC")

# Now, let's put the two plots together
preQC_aggregate_plot <- 
  # Plot the forward and reverse together 
  forward_preQC_plot + reverse_preQC_plot
# Show the plot
preQC_aggregate_plot
```

