---
title: "Pre-processing ASVs with phyloseq"
author: "Daphne Garcia"
date: "2025-03-30"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Set Environment 

## Load Packages, timing, 
```{r load-packages and timing}
#install.packages("BiocManager")
#BiocManager::install("Biostrings")


# Load packages with pacman
# Be sure that you load dada2 & Patchwork, which we didn't do in class on March 12th
pacman::p_load(devtools, phyloseq, dada2, patchwork, Biostrings, tidyverse,
               # Note to add ggpubr & rstatix!!
               ggpubr, rstatix, install = FALSE)

# What time did we start running this script? 
start_time <- Sys.time()
```


## 1. Load Data 

### 1a. Metadata 

```{r load-metadata}
# load in metadata
metadata_df <- 
  read_csv("data/metadata.csv") %>%
  # Fix Column Name
  dplyr::rename("sample_names" = "Run") %>%
  # Add sample names also as a column 
  mutate(names = sample_names) 

# Inspect 
head(metadata_df)
dim(metadata_df)

# include dada2 output
load("data/data_filtered/track_read_counts.RData")

# Take a look
glimpse(track_counts_df)
dim(track_counts_df)

# Check filenames 
head(track_counts_df$sample_names)

# Fix sample names in track_reads 
track_counts_df$sample_names <- sapply(strsplit(track_counts_df$sample_names, "_"), `[`, 1)

# Intuition check 
head(track_counts_df$sample_names)

# What's different? 
setdiff(track_counts_df$sample_names, metadata_df$sample_names)

# Let's do a filtering join with left_join 
metadata_final_df <- 
  metadata_df %>%
  left_join(., track_counts_df, by = "sample_names") %>%
  # sample names to the rownames to merge into phyloseq
  column_to_rownames(var = "sample_names")

# Check 
dim(metadata_final_df)
```

### 1b. ASV Table 

Now, let's load in the ASV count table that we created within DADA2 in `analysis/02_AssignASVs.Rmd`. 

```{r load-asv-table}
asv_df <- 
  read.delim(file = "data/data_filtered/ASV_table.tsv", sep = "\t",
           # add the column names and row names 
           header = TRUE, row.names = 1) 
  #   I'm not including `%>% dplyr::select(-"CJ.V08.P") `because my data doesn't have that, or a ctrl column

# Inspect 
asv_df[1:3, 1:3]

# fix Column names 
## Remove the X: denote at the beginning "^"
colnames(asv_df) <- sub(pattern = "^X", replacement = "" , colnames(asv_df))
## Replace the . with a -: "\\." since . is a special character in regex
colnames(asv_df) <- gsub(pattern = "\\.", replacement = "-", colnames(asv_df))

# Final inspection 
head(colnames(asv_df))
asv_df[1:3, 1:3]
```

### 1c. Taxonomy Table

And, let's also load the taxonomy table that we created in `homework 2: 02_AssignASVs.Rmd`.

```{r load-tax-table}
tax_df <- 
  read.delim("data/data_filtered/ASV_taxonomy.tsv", sep = "\t",
           header = TRUE, row.names = 1) 

# Inspect
dim(tax_df)
dim(asv_df)

# Add check 
stopifnot(rownames(asv_df) == rownames(tax_df))
```


# 2. Handoff to phyloseq 

This is where we will combine our metadata, asv count table, and taxonomy table into a single data S4 data object in R, which we will call `raw_physeq` for now.

```{r phyloseq-handoff}
raw_physeq <- 
  phyloseq(otu_table(asv_df, taxa_are_rows = TRUE),
         sample_data(metadata_final_df),
         tax_table(as.matrix(tax_df)))

# Check out 
raw_physeq

# save 
save(raw_physeq, file = "data/03_PreProcessing/raw_physeq.RData")
```


## 3. Remove chloroplasts

```{r rm-chloro}
noChloros_physeq <- 
  raw_physeq %>%
  subset_taxa(Order != "Chloroplast" | is.na(Order))

# How many ASVs were chloroplasts? 
numChloros_ASVs <- ntaxa(raw_physeq) - ntaxa(noChloros_physeq)
print(numChloros_ASVs)
```

There were `r numChloros_ASVs` ASVs that were chloroplasts. (39)

## 4. Remove Mitochondria 

```{r rm-mitos}
noChlorosMitos_physeq <-  
  noChloros_physeq %>%
  subset_taxa(Family != "Mitochondria" | is.na(Family)) 

# How many ASVs were mitochondria? 
numMitos_ASVs <- ntaxa(noChloros_physeq) - ntaxa(noChlorosMitos_physeq)
print(numMitos_ASVs)
```

There were `r numMitos_ASVs` ASVs that were mitochondria in the data set. (1)

INTERPRETATION #1: There were 39 chloroplast DNA and 1 mitochondrial DNA in the 
dataset. This was honestly surprising, as I thought there would be more 
contamination, given that the dataset already isn't the cleanest. I was also 
surprised that there was more chloroplast DNA than mitochondrial, given that this
is a host-microbiome study. I feel like the amount of contamination (40 total) 
in a dataset of 17,00+ ASVs is very reasonable, and I feel confident in moving
forward. 

# Evaluate and remove the control samples 

Make decision about whether or not to remove the ASVs that we found in our controls:
  1. Negative controls 
  2. ASVs found within the negative controls and their distribution in the samples. 
  3. Evaluate the mock community 
  
  
## 5. Create negative control phyloseq object 

INTERPRETATION #2: My dataset unfortunately did not have any negative controls. 
There are 60 samples reported in the paper (5 replicates of 12 experimental groups).
And for the ASV data frame, there are exactly 60 samples. The paper cited that there 
are two expeirmental groups (gut microbiome at hot temperature, and low temperature), 
and one control group (gut microbiome at control temperature). This does not count 
as a sequencing control because they are gut microbiomes from completely different 
individuals, and do not count as field controls (much less DNA or PCR blanks). 
This is a major bias of this paper.


## 6. Positive Controls 

INTERPRETATION #3:

My dataset does not have positive controls, in addition to not having negative 
controls. They did not add a mock community to the data, so there isn't a positive
control sample that I can compare the rest of my data to.


## 7. Reverse Complements

```{r reverse-complements}
# Pull out Taxa Table
tax_table <- 
  noChlorosMitos_physeq %>%  ### I changed to  noChlorosMitos_physeq from noChlorosMitosNegPosControls_physeq bc I don't have neg or pos ctrls
  tax_table() %>% 
  data.frame()

# Grab ASV Sequences and Names
asv_names <- tax_table$ASV
asv_seqs <- tax_table$ASVseqs

# Create empty comparison matrix
asv_rev_comp <- matrix(nrow=length(asv_seqs), ncol = length(asv_seqs)) 

# Fix the names in the rows and columns 
rownames(asv_rev_comp) <- asv_names
colnames(asv_rev_comp) <- asv_names

# Convert sequences to Biostrings
asv_dna <- DNAStringSet(asv_seqs) 

# Inspect 
head(asv_dna)

# Construct reverse complement
asv_rev_dna <- reverseComplement(asv_dna) 

# Now loop through every asv sequence to check (long step, 2min)
for(i in 1:length(asv_seqs)){ # For each asv sequence...
  match_vec <- asv_dna[i] == asv_rev_dna # Compare it to the reverse complement of every other sequence...
  asv_rev_comp[,i] <- match_vec # Write that logical vector as a new column 
}

# Find how many TRUEs (matches) we have, divide by 2 because each pair occurs twice
cat("For", sum(asv_rev_comp) / 2,"ASVs, the reverse complement will need to be removed") 
```




## 8. Sequencing Depth 

```{r seq-depth}
# The current data object

noChlorosMitosNegPosControls_physeq <- noChlorosMitos_physeq ## bc I don't have neg or pos ctrls
noChlorosMitosNegPosControls_physeq

```

```{r seq-depth pt 2, fig.height=3, fig.width=8 }

# What is the library size/sequencing depth for each sample? 
seqSums_df <- 
  noChlorosMitosNegPosControls_physeq %>%
  otu_table() %>%
  # Sum each sample column 
  colSums() %>%
  data.frame() %>%
  rownames_to_column(var = "names") %>%
  left_join(., metadata_final_df, by = "names") 

# Rename second column 
colnames(seqSums_df)[2] <- "TotalSeqs"

# check
dim(seqSums_df)
head(seqSums_df)

##### chatgpt to make six groups for ggplot out of `Sample Name`
seqSums_df <- seqSums_df %>%
  mutate(sample_name_updated = substr(`Sample Name`, 1, 2))

# Show the depth of samples 
seqSums_df %>%
  dplyr::select(names, TotalSeqs) %>%
  arrange(TotalSeqs) %>%
  head()

# plot it as a bar plot 
numSeq_bar_plot <- 
  seqSums_df %>%
  ggplot(aes(x=reorder(names, TotalSeqs), y = TotalSeqs,
             fill = sample_name_updated)) + 
  geom_bar(stat = "identity") + 
  theme(axis.text.x = element_blank()) + 
  labs(y = "Read Depth", x = "Sample") + 
  theme(legend.position = "none")

# histogram
numSeq_hist_plot <- 
  seqSums_df %>%
  ggplot(aes(x= TotalSeqs, fill =  sample_name_updated)) + 
  geom_histogram(color = "black") + 
  labs(y = "# of Samples", x = "Read Depth") + 
  theme(legend.position = "bottom")

# Density plot 
numSeq_density_plot <- 
  seqSums_df %>%
  ggplot(aes(TotalSeqs, fill =  sample_name_updated)) +
  geom_density(alpha = 0.5) + 
  labs(x = "Read Depth") + 
  theme(legend.position = "none")

# Put it all together 
numSeq_bar_plot + numSeq_hist_plot + numSeq_density_plot + 
  plot_annotation(tag_levels = "A") 
```


NTERPRETATION #5: What can you conclude about your sequencing depths? 

Looking at the bar plot, histogram, and density plot, I think all of the samples 
have sufficient read depth, and are relatively even compared to one another. 

What are your sample distributions? 

Do you have “enough” reads? 
I think all samples have enough reads. Looking at the bar plot, the lowest read 
depth from a sample is slightly under 40,000, but the read depths of samples 
increase evenly, which indicates to me that all of them are real and unalarming.
I would be alarmed if there were samples with no reads, or samples with a 
disproportionately higher read depth than others, but that is not the case with 
this data. 

Are there any samples that should be removed at this step?

The only reads that worry me are the two lowest-read-depth samples in the histogram. 
There are two samples that have a read depth of around 40,000 and 42,000 respectively. 
However, these samples belong to the "high temperature male" and "high temperature 
female" samples. The paper noted that higher temperatures lowered the microbial 
diversity in snail guts, so this may be in line with what the researchers observed too. 

### Remove samples with few reads
```{r min seq depth}
# What's the min seq depth? 
min(sample_sums(noChlorosMitosNegPosControls_physeq))

# Minimum seq depth is 39,242, so I won't be removing any samples
raw_preprocessed_physeq <- noChlorosMitosNegPosControls_physeq
```



## 9. 
```{r save raw_preprocessed_physeq}

save(raw_preprocessed_physeq, file = "data/03_PreProcessing/raw_preprocessed_physeq.RData")
```


# Final info for Reproducibility 

## Check Render Time
```{r stop-time}
# Take the time now that we are at the end of the script
end_time <- Sys.time()
end_time 

# Echo the elapsed time
elapsed_time <- round((end_time - start_time), 3)
elapsed_time
```

